in-progress/7482-sequencer-prover-test-net.md

```diff
@@ -0,0 +1,409 @@
+
+### SPRTN v1.0.0 Sequencer Selection
+
+The validator set will be selected by the multisig.
+
+At the beginning of each epoch, we will assign proposers from the the validator set to slots.
+
+The exact number of validators will be determined via stress tests, modeling, and feedback from the community.
+
+Part of the deliverable for this milestone will be an analysis of network performance based on the validator set size.
+
+### SPRTN v1.0.0 Prover Selection
+
+The list of available provers will be determined by the multisig.
+
+At the beginning of each epoch, a single prover will be selected to generate proofs for the epoch.
+
+### SPRTN v1.0.0 Incentives
+
+For SPRTN v1.0.0, we will create a simple incentives contract within the deployment:
+- Whenever a block is added to the Pending Chain, the proposer will receive a reward.
+
+### The SPRTN v1.0.0 Pending Chain
+
+
+#### Overview
+
+```mermaid
+sequenceDiagram
+    participant PXE as PXE
+    participant L2N as L2 Node
+    participant MP as L2 P2P
+    participant P as L2 Proposer
+    participant V as L2 Validator
+    participant L1R as L1 Rollup Contracts
+
+    P->>L1R: Watches for new L2PendingBlockProcessed events
+    P->>L1R: Download TxEffects
+    P->>P: Apply TxEffects to world state
+    PXE->>L2N: Submit TxObject+Proof
+    L2N->>MP: Add TxObject+Proof
+    MP->>P: Broadcast TxObjects+Proofs
+    MP->>V: Broadcast TxObjects+Proofs
+    P->>P: Execute TxObjects, construct L2 block header
+    P->>MP: Propose TxObjects and L2 block header for a slot 
+    MP->>V: Broadcast proposal
+    V->>V: Execute corresponding TxObjects, construct/compare header
+    V->>MP: Signed proposal (if correct)
+    MP->>P: Aggregate signatures
+    P->>L1R: Submit TxEffects to L1 DA Oracle
+    P->>L1R: Add L2 block to Pending Chain, supplying L2 Header and Signatures as call data
+    L1R->>L1R: Check signatures, verify correct proposer, verify header/TxEffects available
+    L1R->>L1R: Add L2 block to Pending Chain, emit L2PendingBlockProcessed
+```
+
+#### Confirmation Rules
+
+A proposer may submit a block B to the rollup contract on L1 for inclusion in the Pending Chain.
+
+The rollup contract will verify that:
+
+- B is building on the tip of the Pending Chain:
+  - The block number of B is one greater than the block number of the tip of the Pending Chain.
+  - The slot of B is larger than the slot of the tip of the Pending Chain.
+  - The slot of B is in the past
+- B is submitted by the correct proposer for the slot.
+- B contains proof it has signatures from 2/3 + 1 of the committee members attesting to the block.
+- B's header has been made available on L1
+- B's constituent TxEffects have been made available on L1
+
+After this, the block is confirmed in the Pending Chain.
+
+### The SPRTN v1.0.0 Proven Chain
+
+#### Overview
+
+It is the responsibility of the prover selected at the beginning of an epoch to submit the proof of the epoch.
```

Comment:
Do we want to be this strict for version 1? I thought the plan was to push prover coordination for later. Should we initially allow any to submit the proof to keep the chain moving?

Particularly as we can't currently handle re-orgs.

Reply:
This is related to @spalladino and @LHerskind comments below.

How about we keep this *coordination* (we pick a prover for an epoch), but do not worry about any timeliness (so no re-orgs).

In order for the proven chain to advance, all previous proofs must be posted.

Separately, we keep a function on the rollup contract just for verifying whether a proof is valid (not advancing the proven chain), enabling the "free for all" games to be played on SPRTN, DevNet, or both.